source/include/ExecutionContext.h

```diff
@@ -108,15 +110,19 @@ class ExecutionContext
     // Run the concurrent execution system for a short period of time
     ECONTEXT    Int32 /*ExecSlicesResult*/ ExecuteSlices(Int32 numSlices, Int32 millisecondsToRun);
     ECONTEXT    InstructionCore* SuspendRunningQueueElt(InstructionCore* nextInClump);
+    ECONTEXT    void             PauseExecutionContext();
     ECONTEXT    InstructionCore* Stop();
+    ECONTEXT    InstructionCore* Pause(InstructionCore* nextInstruction);
     ECONTEXT    void            ClearBreakout() { _breakoutCount = 0; }
+    ECONTEXT    void            ExecuteAllClumpsTillNextDebugPoint();
     ECONTEXT    void            EnqueueRunQueue(VIClump* elt);
     ECONTEXT    VIClump*        _runningQueueElt;    // Element actually running
     ECONTEXT DebuggingContext* debuggingContext;
  public:
     // Method for runtime errors to be routed through.
     ECONTEXT    void            LogEvent(EventLog::EventSeverity severity, ConstCStr message, ...) const;
-
+    ECONTEXT Int32              getVIPauseState();
```

Comment:
Similar to the comment in ExecutionContext.h, the API should be called getExecutionState since we are dealing with the execution context as a whole and not individual VIs. Also you shouldn't expose the set API.